app/helpers/GoogleData.js

```diff
@@ -3,67 +3,54 @@
  */
 import { GetStoreData, SetStoreData } from '../helpers/General';
 
-function BuildLocalFormat(placeVisit) {
-  return (loc = {
+function formatLocation(placeVisit) {
+  return {
     latitude: placeVisit.location.latitudeE7 * 10 ** -7,
     longitude: placeVisit.location.longitudeE7 * 10 ** -7,
     time: placeVisit.duration.startTimestampMs,
-  });
+  };
 }
 
-function LocationExists(localDataJSON, loc) {
-  let wasImportedBefore = false;
-
-  for (let index = 0; index < localDataJSON.length; ++index) {
-    let storedLoc = localDataJSON[index];
+function hasLocation(localDataJSON, loc) {
+  for (const storedLoc of localDataJSON) {
     if (
-      storedLoc.latitude == loc.latitude &&
-      storedLoc.longitude == loc.longitude &&
-      storedLoc.time == loc.time
+      storedLoc.latitude === loc.latitude &&
+      storedLoc.longitude === loc.longitude &&
+      storedLoc.time === loc.time
     ) {
-      wasImportedBefore = true;
-      break;
+      return true;
     }
   }
 
-  return wasImportedBefore;
-}
-
-function InsertIfNew(localDataJSON, loc) {
-  if (!LocationExists(localDataJSON, loc)) {
-    console.log('Importing', loc);
-    localDataJSON.push(loc);
-  } else {
-    console.log('Existing', loc, localDataJSON.indexOf(loc));
-  }
+  return false;
 }
 
-function Merge(localDataJSON, googleDataJSON) {
-  googleDataJSON.timelineObjects.map(function(
-    data,
-    //index
-  ) {
-    // Only import visited places, not paths for now
-    if (data.placeVisit) {
-      let loc = BuildLocalFormat(data.placeVisit);
-      InsertIfNew(localDataJSON, loc);
-    }
-  });
+function extractNewLocations(storedLocations, googleLocationHistory) {
```

Comment:
Can you add some unit tests for the changes in here? I know there are none yet, but now's a good time to get something there for others to build on.